code/game/machinery/weapon_vending.dm

```diff
@@ -0,0 +1,102 @@
+	icon_state = "weaponry"
+	obj_flags = null //prevents it from being rotated and unanchored
+	machine_name = "gun dispenser"
+	machine_desc = "Holds an internal stock of guns that can be dispensed on-demand or when a charged ID card is swiped, depending on the brand."
+	product_ads = list("Kill some commies for me, will ya?;Warcrimes? HA! What's that?;Hotdogs, Horseshoes & Hand Grenades, America, baby.;'Murica.")
+
+// ## GOC & CI MACHINES //
+
+/obj/machinery/vending/weaponry/goc
+	name = "\improper GOCTech Gun Dispenser v1.1"
+	icon_state = "goc_weaponry"
+	products = list(
+		/obj/item/gun/projectile/automatic/scp/fnfal = 6,
+		/obj/item/gun/projectile/shotgun/pump/combat = 6,
+		/obj/item/gun/projectile/pistol = 6,
+		/obj/item/grenade/frag = 12,
+		/obj/item/ammo_magazine/scp/fnfal = 15,
+		/obj/item/ammo_magazine/scp = 15,
+		/obj/item/ammo_magazine/shot_holder/shell = 2
+	)
+
+/obj/machinery/vending/weaponry/chaos
+	name = "\improper Hacked Gun Dispenser v0.0"
+	desc = "Has outdated software, the auto-update is also busted... Such is the life of a pirate."
+	icon_state = "ci_weaponry"
+	products = list(
+		/obj/item/gun/projectile/automatic/scp/ak74 = 5,
+		/obj/item/gun/projectile/pistol = 5,
+		/obj/item/ammo_magazine/scp/ak = 5
+	)
+
+/obj/machinery/vending/weaponry/chaos/specialized
+	name = "\improper Specialized-Gun Dispenser v0.2"
+	desc = "One of the better devices the Chaos Insurgency has, this one dispenses guns for special operatives."
+	icon_state = "ci_weaponry_special"
+	products = list(
+		/obj/item/gun/projectile/automatic/scp/rpk = 2,
+		/obj/item/gun/projectile/automatic/scp/svd = 2,
+		/obj/item/grenade/frag = 10,
+		/obj/item/ammo_magazine/scp/ak = 30,
+		/obj/item/ammo_magazine/scp/svd = 30
+	)
+
+// ## LCZ MACHINES ## //
+
+/obj/machinery/vending/weaponry/lcz
+	name = "\improper LightCZ AutoEquipment Device"
+	desc = "An automated gun storage device that can dispense guns on demand, this one has been designed to supply LCZ personnel."
+	icon_state = "lcz_weaponry"
+	products = list(
+		/obj/item/gun/projectile/automatic/scp/p90 = 6,
+		/obj/item/gun/projectile/revolver/rhino = 2,
+		/obj/item/gun/projectile/pistol/mk9 = 4
+	)
+
+/obj/machinery/vending/weaponry/lcz/energy
+	name = "\improper LightCZ E.G. Storage"
+	desc = "An automated gun storage device that can dispense guns on demand, this one houses energy/pulse weapons for LCZ personnel."
+	products = list(
+		/obj/item/gun/energy/plasmastun = 1,
+		/obj/item/gun/energy/ionrifle = 1
+	)
+
+// ## HCZ MACHINES ## //
+
+/obj/machinery/vending/weaponry/hcz
+	name = "\improper HContain AutoEquipment Device"
+	desc = "An automated gun storage device that can dispense guns on demand, this one was designed for HCZ personnel."
+	icon_state = "weaponry"
+	products = list(
+		/obj/item/gun/projectile/automatic/scp/p90 = 3,
+		/obj/item/gun/projectile/revolver/rhino = 1,
+		/obj/item/gun/projectile/revolver/mateba = 1,
+		/obj/item/gun/projectile/pistol/mk9 = 1
```

Comment:
the rhino is redundant for HCZ, I made the gun for EZ. 
no magazines or speedloader for the guns?
ammo boxes?

Reply:
fixed